ifopt_ipopt/src/ipopt_solver.cc

```diff
@@ -70,8 +72,15 @@ IpoptSolver::Solve (Problem& nlp)
     throw std::length_error("Ipopt could not initialize correctly");
   }
 
+  // check the jacobian_approximation method
+  bool finite_diff = false;
+  std::string value = "";
+  ipopt_app_->Options()->GetStringValue("jacobian_approximation", value, "");
+  if (value.compare("finite-difference-values") == 0)
+	  finite_diff = true;
+
```

Comment:
how about replacing line 76, 79, 80 with:
```
bool finite_diff = value=="finite-difference-values";
```
keeping declaration and definition closer together. Maybe also rename `value` to something like `jac_type`.

Reply:
Ok, good suggestion.